adapters/types.go

```diff
@@ -0,0 +1,36 @@
+package adapters
+
+import (
+	"fmt"
+	"strings"
+)
+
+type DeleteCondition struct {
+	Field  string
+	Value  interface{}
+	Clause string
+}
+
+func (dc *DeleteCondition) ToQueryString() string {
+	var value string
+	switch dc.Value.(type) {
```

Comment:
we can't trust the value type because there might be explicit type casts for example:
a field might be `integer` but has explicit type `text` in mappings and in DWH.

Reply:
Delete conditions are converted to queries at adapters now, so, type casts and placeholders are used based on type of destination